pkg/cloud/cloud.go

```diff
@@ -18,6 +19,38 @@ import (
 	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud/vsphere"
 )
 
+// cloudConfigTransformer function transforms the source config map using the input infrastructure.config.openshift.io object.
+// only the data and binaryData field of the output ConfigMap will be respected by consumer of the transformer.
+type cloudConfigTransformer func(source *corev1.ConfigMap, infra *configv1.Infrastructure) (*corev1.ConfigMap, error)
+
+
+// GetCloudConfigTransformer returns the function that should be used to transform
+// the cloud configuration config map
+func GetCloudConfigTransformer(platformStatus *configv1.PlatformStatus) (cloudConfigTransformer, error) {
+	switch platformStatus.Type {
+	case configv1.AlibabaCloudPlatformType:
+		return common.NoOpTransformer, nil
+	case configv1.AWSPlatformType:
+		return common.NoOpTransformer, nil
+	case configv1.AzurePlatformType:
+		return common.NoOpTransformer, nil
+	case configv1.GCPPlatformType:
+		return common.NoOpTransformer, nil
+	case configv1.IBMCloudPlatformType:
+		return common.NoOpTransformer, nil
+// TODO: Enable this when we turn on syncing for OpenStack
+//	case configv1.OpenStackPlatformType:
+//	 	return openstack.CloudConfigTransformer, nil
```

Comment:
Are we not doing that within this PR?

Reply:
I've done this now. This was incomplete.